### PR TITLE
Fix #52 Creation of Menu solving Ant export should be a separate export

### LINKED ISSUE #52: Ant export should be a separate export option
You can export a project as "LibreOffice package" (File->Export).
On the last step, you have a checkbox "Save as Ant script", which generates an Ant script to build the extension without Eclipse.

This is a useful feature, and it should be a separate wizard, not hidden where it is currently.
So the user should be able to see "Generate Ant Script" when going to File->Export, next to "LibreOffice package".

### PR DESCRIPTION
**Pull request by shobhanmandal**
A new Wizard Menu under LibreOffice is now shown which allows you to create the ANT file build.xml in one go rather than going for Package Export followed by the ANT file generation.

### CODE REVIEW

**Comment by smehrbrodt** [comment]
When I select the new option in the export dialog, I get:
`The selected wizard could not be started.
Plug-in org.libreoffice.ide.eclipse.core was unable to load class org.libreoffice.ide.eclipse.core.wizards.AntScriptWizard.
org.libreoffice.ide.eclipse.core.wizards.AntScriptWizard cannot be found by org.libreoffice.ide.eclipse.core_2.2.6`

**Comment by smehrbrodt** [comment]
Ok the problem was on my side, the new java files were not compiled somehow.

This works basically :)

Some points:
- Please remove the ant export option from the original export wizard
- The whole ant export should be in the "java" package, not in "core". Everything in core should not be language dependant. We want to have python support for example at some stage. There we don't need ant export.

Also please have a look at your coding style. There was some incorrect indentation and some excessive blank lines. You might consider using Eclipse's source format feature.

File: java/source/org/libreoffice/ide/eclipse/java/export/JavaExportPart.java [hunk 1]
```diff
@@ -66,9 +63,20 @@
...
     private JavaExportPageControl mController;
+    private static AntScriptExportWizardPage mAntScriptPage;
+
+    public static final int HORIZONTAL_INDENT = 20;
+
+    /**
+     * @param pAntScriptPage
+     *            Helps in retrieving the project selected in this part.
+     */
+    public static void setAntScriptExportPage(AntScriptExportWizardPage pAntScriptPage) {
```
Comment (smehrbrodt) [inline java/source/org/libreoffice/ide/eclipse/java/export/JavaExportPart.java:77]: Please no references to the ant stuff from here. If you need a shared method, add it to some helper class or something.

File: core/source/org/libreoffice/ide/eclipse/core/wizards/pages/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/UnoPackageExportPage.java
```
Comment (smehrbrodt) [inline core/source/org/libreoffice/ide/eclipse/core/wizards/pages/.gitignore]: Why this?

File: core/source/org/libreoffice/ide/eclipse/core/wizards/pages/UnoPackageExportPage.java [hunk 3]
```diff
@@ -145,7 +145,7 @@ private void loadData() {
             i++;
         }
 
-        if(selected) {
+        if (selected) {
```
Comment (smehrbrodt) [inline core/source/org/libreoffice/ide/eclipse/core/wizards/pages/UnoPackageExportPage.java:148]: Is this whole file still needed?

File: core/plugin.properties [hunk 4]
```diff
@@ -14,6 +14,7 @@ wizards.service          = UNO service
 wizards.interface        = UNO interface
 wizards.export			 = LibreOffice package
 
+
```
Comment (smehrbrodt) [inline core/plugin.properties]: Please no whitespace / new line changes in existing files.

File: core/source/org/libreoffice/ide/eclipse/core/model/language/LanguageExportPart.java [hunk 5]
```diff
@@ -78,4 +78,5 @@ public void setPage(ManifestExportPage pPage) {
     protected ManifestExportPage getPage() {
         return mPage;
     }
+
```
Comment (smehrbrodt) [inline core/source/org/libreoffice/ide/eclipse/core/model/language/LanguageExportPart.java]: Same here

**Review by smehrbrodt — CHANGES_REQUESTED** [review]

File: core/source/org/libreoffice/ide/eclipse/core/wizards/pages/.gitignore (hunk 2 above)
Comment (shobhanmandal) [reply core/source/org/libreoffice/ide/eclipse/core/wizards/pages/.gitignore]: Sorry, this file got created and pushed, would remove it

File: core/source/org/libreoffice/ide/eclipse/core/wizards/pages/UnoPackageExportPage.java (hunk 3 above)
Comment (shobhanmandal) [reply core/source/org/libreoffice/ide/eclipse/core/wizards/pages/UnoPackageExportPage.java:148]: Yeah, this file forms the first page of the Export -> 'LibreOffice Package', the Ant Script gets build from the 'ManifestExportPage' which is the second page in the export. Nothing related to Ant Script building was form the first page, so has to be kept intact, second page 'ManifestExportPage' though had a number of changes.

File: java/source/org/libreoffice/ide/eclipse/java/export/AntScriptExportWizardPage.java [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+    @Override
+    public void createControl(Composite pParent) {
+
+        Composite body = new Composite(pParent, SWT.NONE);
+        body.setLayout(new GridLayout());
+        body.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+        setControl(body);
+
+        createProjectSelection();
+        sContentSelector = new PackageContentSelector(body, SWT.NONE);
```
Comment (smehrbrodt) [inline java/source/org/libreoffice/ide/eclipse/java/export/AntScriptExportWizardPage.java]: I think we don't need the PackageContentSelector on this wizard. It's not used in the ant export.

File: java/source/org/libreoffice/ide/eclipse/java/export/AntScriptExportWizard.java [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                    section = workbenchSettings.addNewSection(DIALOG_SETTINGS_KEY);
+                    setDialogSettings(section);
+                }
+
+            } catch (Exception e) {
+                PluginLogger.error("The Ant Script couldn't be built", e);
+            }
+        }
+
+        File tmpDir = new File(directory + "/build.xml");
```
Comment (smehrbrodt) [inline java/source/org/libreoffice/ide/eclipse/java/export/AntScriptExportWizard.java]: Please chose a better name than tmpDir, maybe antFile?

File: java/source/org/libreoffice/ide/eclipse/java/export/JavaExportPart.java [hunk 8]
```diff
@@ -151,10 +132,24 @@ public void dispose() {
      */
     @Override
     public void doFinish(UnoPackage pModel) {
-        if (mController.getSaveAntScript()) {
+
+        String directory = mAntScriptPage.getPath();
+        File tmpDir = new File(directory + "/build.xml");
```
Comment (smehrbrodt) [inline java/source/org/libreoffice/ide/eclipse/java/export/JavaExportPart.java]: Please chose a better name than tmpDir, maybe antFile?

File: java/source/org/libreoffice/ide/eclipse/java/export/AntScriptExportWizard.java [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+import org.libreoffice.ide.eclipse.core.model.ProjectsManager;
+import org.libreoffice.ide.eclipse.java.Messages;
+import org.libreoffice.plugin.core.model.UnoPackage;
+
+/**
+ * Class for the new Ant Script Generation wizard.
+ */
+public class AntScriptExportWizard extends Wizard implements IExportWizard {
+
+    private static final String DIALOG_SETTINGS_KEY = "oxt.export"; //$NON-NLS-1$
```
Comment (smehrbrodt) [inline java/source/org/libreoffice/ide/eclipse/java/export/AntScriptExportWizard.java]: Please chose a different key here. this one is used for the package export.

File: core/source/org/libreoffice/ide/eclipse/core/wizards/pages/ManifestExportPage.java [hunk 10]
```diff
@@ -118,7 +116,7 @@ public ManifestExportPage(String pPageName, IUnoidlProject pProject) {
      */
     public void setProject(IUnoidlProject pProject) {
         mProject = pProject;
-        reloadLanguagePart();
+        //        reloadLanguagePart();
```
Comment (smehrbrodt) [inline core/source/org/libreoffice/ide/eclipse/core/wizards/pages/ManifestExportPage.java]: Is this still needed? If not please remove.

File: java/source/org/libreoffice/ide/eclipse/java/export/JavaExportPart.java [hunk 11]
```diff
@@ -66,9 +63,20 @@
     private Button mSaveScripts;
```
Comment (smehrbrodt) [inline java/source/org/libreoffice/ide/eclipse/java/export/JavaExportPart.java]: This button seems unused?

File: java/source/org/libreoffice/ide/eclipse/java/export/AntScriptExportWizardPage.java [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+import org.libreoffice.ide.eclipse.core.model.IUnoidlProject;
+import org.libreoffice.ide.eclipse.core.model.ProjectsManager;
+import org.libreoffice.ide.eclipse.core.model.language.LanguageExportPart;
+import org.libreoffice.ide.eclipse.java.Messages;
+import org.libreoffice.plugin.core.model.UnoPackage;
+
+public class AntScriptExportWizardPage extends WizardPage {
+
+    private IUnoidlProject sSelectedProject;
+    private LanguageExportPart sLangPart;
```
Comment (smehrbrodt) [inline java/source/org/libreoffice/ide/eclipse/java/export/AntScriptExportWizardPage.java:60]: Hm this LanguageExportPart was designed to add some language specific options to the package export wizard.
It seems a bit weird that we have a LangaugeExportPart on a java specific AntScriptWizardPage.
Maybe you can rename it so it matches better what it's actually used for now? Also I don't know if we still need to derive from the LanguageExportPart, or we just make it an own class?
Maybe also just include the relevant stuff directly here without this extra class?

Not sure what is best, but this looks cumbersome at least.

**Review by smehrbrodt — CHANGES_REQUESTED** [review]

**Pull request merged**